FEATURES.md

```diff
@@ -0,0 +1,97 @@
+
+From this point, the maintainers will engage you on the feature proposal in the issue comments. The goal of this conversation is to clarify if the feature proposal meets the criteria of a new feature in the project. An additional goal is to determine if this is something that you, as the submitter, can contribute yourself or if you’ll need assistance from additional contributors.
+
+The criteria to continue with a feature proposal are as follows:
+
+- __Is the proposal describing a feature?__
+ The feature proposal isn’t meant for bug fixes, refactoring, or refinement to existing features. Sometimes this line can be fuzzy so the maintainers will help clarify.
+- __Does this feature fit in the scope of the repo and project?__
+Some things might be good ideas but they might not fit either with the project as a whole in the specific repo in which the issue was posted. In the latter case, the maintainers will help you determine the right home and transfer the issue.
+- __Is the proposed feature sufficiently distinct from existing features?__
+With a project like OpenSearch it’s totally normal to not know every capability, so this criteria is designed to suss out if the feature provides unique capabilities when compared to existing functionality.
+- __Does the proposed feature overlap or conflict with ongoing work?__
+The best use of everyone’s time is to ensure that there are no competing implementations of features, so this criteria establishes that there is not more than one effort going on for any given area.
+- __Is the feature proposal sufficiently atomic as a feature?__
+Sometimes big ideas should be broken up into smaller ideas. Complete atomicity is probably unobtainable given interdependencies between different facets of the product however the goal here is to make sure that the proposed features are accomplishable in a reasonable amount of time or if it would make more sense to break up the feature into smaller chunks.
+
+The maintainers of the repo decide when and if the proposal meets the criteria. If the feature proposal doesn’t meet the criteria, then the issue will be closed with an explanation and tagged clearly. If the feature does meet the criteria and someone is volunteering to work on the feature, it moves to the next stage. If the feature still needs someone to write the feature, it is tagged as “help wanted” until someone volunteers.
+
+In this process your issue might be closed for a variety of reasons and it doesn't necessarily mean that it’s a bad idea or that the feature isn’t wanted. It might mean that the way it is presented, supported, or the context of the other development in the project doesn’t align with the feature proposal process. If this is the case, the maintainer can help with next steps to move forward.
+
+__Design Document__
```

Comment:
I personally strongly disagree with any guidelines on the design document. This is a whole other proposal which implies designs are written in markdown and are stored in repos. We don't do this today, and I think this should be a separate question.

Reply:
@dblock is the issue that this is perceived as rigid? My understanding was the spirit of this was to have a baseline anyone could build off of, but it's okay to adapt/adjust it as needed.

Reply:
I would prefer we discuss the need for a design document as a subsequent PR. I have a lot to say about it :) Would it be okay to reduce this PR to things that are uncontroversial and make progress incrementally?

Reply:
The "design document" is not really defined; hence there is no point in making something like this as required. They should just make a proposal with details and the rest makes sense.

Reply:
> I am open to postponing the discussion on design doc formatting/template guidelines. But I do feel it's necessary.

Exactly. I am not saying it's not necessary. I'm just saying let's start with a new FEATURES.md that has a bare minimum talking about features.